keps/sig-testing/1746-stage-e2e-test-framework/README.md

```diff
@@ -0,0 +1,610 @@
+  without having to deal with the problems of importing the entirety of core
+  Kubernetes.
+
+There are two main components to the e2e test framework: its configuration
+(used for setting up and running e2e tests) and its set of packages and utility
+functions.
+For this work, we plan on focusing on the latter.
+The entrypoint into the framework, the configuration part, will stay as is.
+Only the utility functions will be refactored.
+What this means for consumers, is that the functions offered via the e2e test
+framework will change but not the way e2e tests are setup and executed.
+
+### Goals
+
+<!--
+List the specific goals of the KEP.  What is it trying to achieve?  How will we
+know that this has succeeded?
+-->
+
+* Decouple the e2e test framework and all of its subpackages from any
+  Kubernetes internal packages (i.e., pkg, other e2e packages).
+* Impose import restrictions to prevent the use of core Kubernetes packages.
+* Allow the e2e test framework to be easily used and without requiring the
+  entirety of Kubernetes to be imported.
+* Refactor the e2e test framework so that its sub-packages are well contained
+  (i.e., `framework/pod/*` contains all necessary utilities to manage pods
+  during e2e tests).
+  * Update Kubernetes e2e tests to use the refactored version of any utility
+    function we change.
+
+### Non-Goals
+
+* Change the way e2e tests/jobs are setup or executed.
+* This enhancement is not directed at making the e2e test framework
+  provider-agnostic. Cloud provider functionality will be kept as is unless it
+  is used in inadequate e2e test framework sub-packages.
+* The e2e test framework will be refactored as needed strictly to remove its
+  dependence from other internal Kubernetes packages. It is not the goal of
+  this enhancement to propose an entire redesign of the e2e test framework.
+
+<!--
+What is out of scope for this KEP?  Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.  The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+In order to cleanup and refactor the e2e test framework we will have to
+decouple the e2e test framework from internal Kubernetes packages and to
+organize test code in e2e tests and in the e2e test framework.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+I, as a core Kubernetes contributor, want a e2e test framework that is well
+organized and containsa core of functions that will help me write clean and
+maintainable e2e tests.
```

Comment:
The non-goals above suggest that this is not the case.

nit: missing a space after contains.

Reply:
Thank you for the catch and for the comment!

Added a bit on the non-goals section to call out that we are not proposing a redesign of the core of the e2e test framework, the core being the test context and how it is used.